src/main/java/com/assignment1/parser/Validator.java

```diff
@@ -0,0 +1,157 @@
+package com.assignment1.parser;
+
+import com.assignment1.Type;
+import com.assignment1.exceptions.RuntimeExceptionCustom;
+import com.assignment1.item.ItemEntity;
+
+/**
+ * Class to implement Valid Interface for validation methods.
+ */
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+public class Validator implements Valid {
+
+  /**
+   * Custom Exception class object.
+   */
+  private final transient RuntimeExceptionCustom customException =
+          new RuntimeExceptionCustom("Invalid Arguments Exception");
+
+  /**
+   * Checks the format according to option(name, price, quantity, type.
+   */
+  public boolean checkFormat(final String option, final String str) {
+
+    boolean result = false;
+
+    //Price can have one decimal point and only digits, no minus symbol (negative)
+    //Price value length <= 10
+    if ("-price".equals(option)) {
+      result = str.indexOf('.') == str.lastIndexOf('.')
+              && str.matches("[0-9.]*") && str.length() <= 10;
+    }
```

Comment:
- it can have commas in it
- why is there a limit on length?

better to just have  check that cover all types

Reply:
maybe just a regex is better that multiple string operations

Reply:
Used the regex to handle the commas for Indian currency pattern. 
Limit on length is removed. 
Used only the regex now, avoided string operations.